### PR TITLE
added solution

### PR DESCRIPTION
**Pull request by floriangeyer21**

### CODE REVIEW

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        User userFromDB = userService.findByLogin(login).orElseThrow(() ->
+                new AuthenticationException("Incorrect users name or password"));
+
+        if (userFromDB.getPassword().equals(password)) {
```
Comment (dariiabutorina) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]:
```suggestion
                new AuthenticationException("Incorrect users name or password"));
        if (userFromDB.getPassword().equals(password)) {
```

File: src/main/webapp/WEB-INF/views/login.jsp [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+            text-align: center;
+        }
+        .font1 {
+            font-size: 120%;
+        }
+        a:focus {
+            background-color: red;
+        }
+
+        .body1 {
```
Comment (dariiabutorina) [inline src/main/webapp/WEB-INF/views/login.jsp]:
```suggestion
        }
        .body1 {
```

File: src/main/webapp/WEB-INF/views/login.jsp [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+<h4 style="color:red" class="font : font1">${errMessage}</h4>
+<form method="post" action="${pageContext.request.contextPath}/login">
+    <h1 class="font">Sign in</h1>
+    <div class="login">
+        <p>Please fill in this form to create an account.</p>
+        <hr>
+        <label for="login"><b>Login</b></label>
+        <input type="text" placeholder="Enter login" name="login" required="required" value=${login}>
+
+        <label for="psw"><b>Password</b></label>
```
Comment (dariiabutorina) [inline src/main/webapp/WEB-INF/views/login.jsp]:
```suggestion
        <input type="text" placeholder="Enter login" name="login" required="required" value=${login}>
        <label for="psw"><b>Password</b></label>
```

File: src/main/webapp/WEB-INF/views/login.jsp [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+        <input type="password" placeholder="Enter Password" name="pwd" required>
+        <hr>
+
+        <button type="submit" class="/login">Sign in</button>
```
Comment (dariiabutorina) [inline src/main/webapp/WEB-INF/views/login.jsp]:
```suggestion
        <hr>
        <button type="submit" class="/login">Sign in</button>
```

File: src/main/webapp/WEB-INF/views/login.jsp [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+    </div>
+</form>
+
+
+<div class="font">
```
Comment (dariiabutorina) [inline src/main/webapp/WEB-INF/views/login.jsp]:
```suggestion
</form>
<div class="font">
```

File: src/main/webapp/WEB-INF/views/login.jsp [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+    <a href="${pageContext.request.contextPath}/">To main page</a>
+    <h5> or </h5>
+    <a href="${pageContext.request.contextPath}/registration">Create new account</a>
+</div>
+</body>
+</html>
+
```
Comment (dariiabutorina) [inline src/main/webapp/WEB-INF/views/login.jsp:51]:
```suggestion
</html>
```

**Review by dariiabutorina — CHANGES_REQUESTED** [review]

File: src/main/java/com/internet/shop/controllers/LoginController.java [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+
+    @Override
+    protected void doPost(HttpServletRequest req, HttpServletResponse resp)
+            throws ServletException, IOException {
+        String login = req.getParameter("login");
+        String pwd = req.getParameter("pwd");
+        try {
+            User user = authenticationService.login(login, pwd);
+            HttpSession session = req.getSession();
+            session.setAttribute("user_id", user.getId());
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/LoginController.java]: let's make user_id a constant

File: src/main/java/com/internet/shop/controllers/user/GetAllUsersController.java [hunk 8]
```diff
@@ -12,6 +12,7 @@
 
 public class GetAllUsersController extends HttpServlet {
     private static final Injector injector = Injector.getInstance("com.internet.shop");
+    private static final String USER_ID = "user_id";
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/controllers/user/GetAllUsersController.java]: you are not using it here

File: src/main/java/com/internet/shop/security/AuthenticationService.java [hunk 9]
```diff
@@ -0,0 +1,8 @@
+package com.internet.shop.security;
+
+import com.internet.shop.model.User;
+
+public interface AuthenticationService {
+    User login(String login, String password)
+            throws com.internet.shop.exceptions.AuthenticationException;
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationService.java]: let's use import

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,25 @@
...
+            return userFromDB;
+        } else {
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: do you need else after return?

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java (hunk 1 above)
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: either make both checks in if() or make exception message a constant

File: src/main/java/com/internet/shop/controllers/user/GetAllUsersController.java (hunk 8 above)
Comment (dmytro-hryhoruk) [reply src/main/java/com/internet/shop/controllers/user/GetAllUsersController.java]: ?

File: src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java [hunk 11]
```diff
@@ -0,0 +1,23 @@
...
+@Service
+public class AuthenticationServiceImpl implements AuthenticationService {
+    @Inject
+    private UserService userService;
+
+    @Override
+    public User login(String login, String password) throws AuthenticationException {
+        if (userService.findByLogin(login).isPresent()
+                && userService.findByLogin(login).get().getPassword().equals(password)) {
+            User userFromDB = userService.findByLogin(login).get();
```
Comment (dmytro-hryhoruk) [inline src/main/java/com/internet/shop/security/AuthenticationServiceImpl.java]: you are calling findByLogin() three times. Search through a database it's an expensive operation, let's save the result to local variable

File: src/main/java/com/internet/shop/controllers/user/GetAllUsersController.java (hunk 8 above)
Comment (dmytro-hryhoruk) [reply src/main/java/com/internet/shop/controllers/user/GetAllUsersController.java]: @floriangeyer21 do you need this constant here?

**Review by dmytro-hryhoruk — APPROVED** [review]

**Pull request merged**